public/components/map_container/map_container.tsx

```diff
@@ -68,6 +73,25 @@ export const MapContainer = ({
     maplibreInstance.on('move', () => {
       return setZoom(Number(maplibreInstance.getZoom().toFixed(2)));
     });
+    const mapContainerElement: HTMLElement | null = document.querySelector('.map-page');
+    let resizeChecker: ResizeChecker;
+    if (mapContainerElement) {
+      resizeChecker = new ResizeChecker(mapContainerElement);
+      if (inDashboardMode) {
+        resizeChecker.on(
+          'resize',
+          throttle(() => {
+            maplibreInstance?.resize();
```

Comment:
Too many control statements. Why not create new component for Dashboard?

Reply:
Creating new component for dashboards will make many duplicated component and logics

Reply:
Yes, i am hoping that we can reuse the components, not logic.